__tests__/commands/install/focus.js

```diff
@@ -224,6 +224,30 @@ test.concurrent('focus works correctly when focusing on a scoped package', (): P
   });
 });
 
+test.concurrent('focus work even when the workspace root also depends on other workspaces', (): Promise<void> => {
+  return runInstall(
+    {focus: true},
+    {source: 'focus-root-dep', cwd: '/packages/example-yarn-workspace-1'},
+    async (config, reporter) => {
+      const existAndIsMaterialized = async moduleDir => {
+        expect(await fs.exists(path.join(moduleDir, 'package.json'))).toBeTruthy();
+        const stat = await fs.lstat(moduleDir);
+        expect(stat.isSymbolicLink()).toEqual(false);
+      };
+
+      // The one in the root must exist, and must be materialized (since we're focusing on example-yarn-workspace-1)
+      await existAndIsMaterialized(path.join(config.cwd, '..', '..', 'node_modules', 'example-yarn-workspace-2'));
+
+      // If there's one in example-yarn-workspace-1, it also has to be materialized
+      if (await fs.exists(path.join(config.cwd, 'node_modules', 'example-yarn-workspace-2'))) {
```

Comment:
there definitely should be one at this path, so I would recommend removing this check.

Reply:
Ideally there shouldn't, because it should be hoisted to the root (since both the root and `example-yarn-workspace-1` share the same dependency to `example-yarn-workspace-2`).

Reply:
see my comment above. It would never hoist the remote copy of the workspace.

Reply:
oh i see what you're trying to do. You want to allow the stuff in the root workspace to depend on the pre-built copy of example-yarn-workspace-2.

Making something like that work would be a much bigger change I think. Because focus was only intended to affect the workspace you focus on. So the root workspace was never really intended to be affected. Also I didn't realize the root workspace was a regular workspace that could be used but that's a separate thing.